app/backend/gwells/middleware/GWellsRequestParsingMiddleware.py

```diff
@@ -1,11 +1,36 @@
-#attribution https://thihara.github.io/Django-Req-Parsing/
+# attribution https://thihara.github.io/Django-Req-Parsing/
+import time
+from random import randint
+import logging
 
 from django.http import HttpResponseBadRequest
 from django.utils.deprecation import MiddlewareMixin
 from django.http import HttpResponse
 
+from gwells.settings.base import get_env_variable
+
+
+logger = logging.getLogger(__name__)
+
+
 class GWellsRequestParsingMiddleware(MiddlewareMixin):
 
+    def latency(self, request):
+        """
+        As a developer, one is blind to various issues that may crop up due to network latency.
+        Set SIMULATE_NETWORK_LATENCY=True in your environment to introduce all manner of delays to frustrate
+        you.
+        """
+        SIMULATE_NETWORK_LATENCY = get_env_variable('SIMULATE_NETWORK_LATENCY',
+                                                    default_value=False,
+                                                    strict=False,
+                                                    warn=False)
+        if SIMULATE_NETWORK_LATENCY == 'True':
```

Comment:
minor quibble but if this is a long-term addition, can SIMULATE_NETWORK_LATENCY live in the `gwells.settings` file as
```
        SIMULATE_NETWORK_LATENCY = get_env_variable('SIMULATE_NETWORK_LATENCY',
                                                    default_value=False,
                                                    strict=False,
                                                    warn=False) == 'True'
```
and import the settings.SIMULATE_NETWORK_LATENCY boolean here?

Reply:
You could make it it's own middleware and then in the settings.py file do:
```
if SIMULATE_NETWORK_LATENCY:
    MIDDLEWARE = MIDDLEWARE + [ NetworkDelayMiddleware ] 
```
that way the network delay middleware will not be included in the middleware chain for requests on prod.

Semi-related note...it would be good to remove GwellsRequestParsingMiddleware.

Reply:
I've created a ticket for GwellsRequestParsingMiddleware.

Delays should not be in the code. My solution, using nginx as reverse_proxy on local machine with a random delay.

brew install nginx-full --with-echo-module --with-lua-module
edit /usr/local/etc/nginx/nginx.conf ; nginx listening on port 8000, forwarding traffic to gwells running on 8001, with a random delay on the api:
`
        location /gwells/api/ {
           set $delay 0;
           rewrite_by_lua_block {
              ngx.var.delay = math.random(0, 5)
           }
           echo_sleep $delay;
           echo_exec @api;
        }

        location / {
           proxy_pass http://127.0.0.1:8001;
        }

        location @api {
            proxy_pass http://127.0.0.1:8001;
        }
`